ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
+        # cleanup
+        # -------
         self.delete(ENTITY, [self.data])
         self.delete(hostgroup.ENTITY, [hostgroup.DATA, hostgroup.DATA2])
         self.delete(netgroup.ENTITY, [netgroup.DATA])
         self.delete(rbac.ROLE_ENTITY, [rbac.ROLE_DATA])
         self.delete(hbac.RULE_ENTITY, [hbac.RULE_DATA])
         self.delete(sudo.RULE_ENTITY, [sudo.RULE_DATA])
+
+    @screenshot
+    def test_buttons(self):
+        """Test buttons"""
+        self.init_app()
+        actions = ActionChains(self.driver)
+        self.navigate_to_entity(ENTITY)
+        # add with enter key
+        self.navigate_to_entity(ENTITY)
+        self.button_click('add')
+        self.fill_text('div[name="hostname"] input[name="hostname"]',
+                       self.pkey)
+        self.check_option('force')
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait_for_request(n=3)
+        self.assert_record(self.pkey)
+
+        # add and another
+        self.add_record(ENTITY, [self.data2, self.data3])
+        # webui automatically lower case
+        self.pkey3 = self.pkey3.lower()
+        self.data3['pkey'] = self.data3['pkey'].lower()
+        self.pkey4 = self.pkey4.lower()
+        self.data4['pkey'] = self.data4['pkey'].lower()
+
+        # add and edit record
+        self.add_record(ENTITY, self.data4, dialog_btn='add_and_edit')
+        self.assert_facet(ENTITY, facet="details")
+
+        # cancel managedby
+        self.add_associations([self.pkey3], facet='managedby_host',
+                              confirm_btn="cancel")
+        self.wait()
+        fqdn_table = self.find('table[name="fqdn"]', By.CSS_SELECTOR)
+        self.select_record(self.pkey4, parent=fqdn_table)
+        self.button_click('remove')
+        self.dialog_button_click('cancel')
+        self.assert_record(self.pkey4)
+
+        # add duplicate
+        self.add_record(ENTITY, self.data2, negative=True, pre_delete=False)
+        dialog_info = self.get_dialog_info()
+        expected_msg = 'host with name "' + self.data2['pkey'] + \
+                       '" already exist'
+        if expected_msg in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+        else:
+            assert False, "Duplicate dialog missing or have wrong text."
+
+        # duplicate with pressed keys
+        self.add_record(ENTITY, self.data2, negative=True, pre_delete=False)
+        actions.send_keys(Keys.TAB).perform()
+        self.wait()
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait(3)
+        self.dialog_button_click('cancel')
+        self.assert_no_dialog()
+
+        # remove multiple
+        self.navigate_to_entity(ENTITY)
+        self.select_multiple_records([self.data2, self.data3, self.data4])
+        self.facet_button_click('remove')
+        self.wait()
+        self.check_option('updatedns')
+        self.dialog_button_click('ok')
+        self.assert_notification()
```

Comment:
Maybe it would be good to provide text the tests needs to assert. Or is any "success" OK? The same  later.

Reply:
as function has default type success it is enough to have success notification as other notification are already closed.